src/binary-search-tree.js

```diff
@@ -5,6 +5,31 @@ class BinarySearchTree {
     this.left = null;
     this.right = null;
   }
+  insert(value) {
+    const node = new BinarySearchTree(value);
+    const recFunction = (obj) => {
+      if (value < obj.value && obj.left === null) obj.left = node;
+      else if (value < obj.value) recFunction(obj.left);
+      else if (value > obj.value && obj.right === null) obj.right = node;
+      else if (value > obj.value) recFunction(obj.right);
+    };
+    recFunction(this);
+  }
+  contains(value) {
+    let bool = false;
+    const recFunction = (obj) => {
+      if (value === obj.value) bool = true;
+      else if (value < obj.value && obj.left !== null) recFunction(obj.left);
+      else if (value > obj.value && obj.right !== null) recFunction(obj.right);
+    };
+    recFunction(this);
+    return bool;
+  }
+  depthFirstForEach(cb) {
+    cb(this.value);
+    if (this.left !== null) this.left.depthFirstForEach(cb);
```

Comment:
Here you can just do `if (this.left)` and `if (this.right)`. No need to explicitly check that they aren't `null`.